packages/shared/src/components/modals/SharedBookmarksModal.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import request from 'graphql-request';
+import CopyIcon from '../../../icons/copy.svg';
+import { Button } from '../buttons/Button';
+import { ResponsiveModal } from './ResponsiveModal';
+import { ModalProps } from './StyledModal';
+import styles from './AccountDetailsModal.module.css';
+import { Switch } from '../fields/Switch';
+import {
+  BookmarksSharingData,
+  BOOKMARK_SHARING_MUTATION,
+  BOOKMARK_SHARING_QUERY,
+} from '../../graphql/bookmarksSharing';
+import { apiUrl } from '../../lib/config';
+import { TextField } from '../fields/TextField';
+import TwitterIcon from '../../../icons/twitter_color.svg';
+import SlackIcon from '../../../icons/slack.svg';
+import DiscordIcon from '../../../icons/discord_color.svg';
+import GithubIcon from '../../../icons/github.svg';
+import { ModalCloseButton } from './ModalCloseButton';
+import { useCopyLink } from '../../hooks/useCopyLink';
+
+export default function SharedBookmarksModal({
+  className,
+  ...props
+}: ModalProps): ReactElement {
+  const { data: bookmarksSharingData, isFetched } =
+    useQuery<BookmarksSharingData>('bookmarksSharing', () =>
+      request(`${apiUrl}/graphql`, BOOKMARK_SHARING_QUERY),
+    );
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: updateBookmarksSharing } = useMutation<{
+    enabled: boolean;
+  }>(
+    () => {
+      const updatedValue = !bookmarksSharingData?.bookmarksSharing?.enabled;
+      return request(`${apiUrl}/graphql`, BOOKMARK_SHARING_MUTATION, {
+        enabled: updatedValue,
+      });
+    },
+    {
+      onSuccess: (data) => {
+        queryClient.setQueryData('bookmarksSharing', data);
+      },
+    },
+  );
+
+  const [isRssUrlCopied, copyRssUrl] = useCopyLink(
+    () => bookmarksSharingData?.bookmarksSharing?.rssUrl,
+  );
+
+  if (!isFetched || !bookmarksSharingData?.bookmarksSharing) {
+    return <></>;
+  }
+
+  return (
+    <>
+      <ResponsiveModal
+        className={classNames(className, styles.accountDetailsModal)}
+        {...props}
+      >
+        <header className="flex items-center justify-between py-4 px-6 w-full border-b border-theme-divider-tertiary">
+          <h3 className="pl-2 font-bold typo-title3">Bookmarks sharing</h3>
+          <ModalCloseButton onClick={props.onRequestClose} />
+        </header>
+        <section className="flex flex-col py-6 px-6 mobileL:px-10">
+          <Switch
+            inputId="share-bookmarks-switch"
+            name="hide-read"
```

Comment:
Not sure if this name reflects what it does, perhaps we can name it `share-bookmarks`
What do you think?

Reply:
Good catch.